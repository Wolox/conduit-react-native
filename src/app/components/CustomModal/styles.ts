import { StyleSheet } from 'react-native';
import { darkGray, white } from '@constants/colors';

const COMMON_MARGIN = 16;
const COMMON_PADDING = 20;

const styles = StyleSheet.create({
  container: {
    width: '90%',
    minHeight: 120,
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: white,
    borderRadius: 16
  },
  exitButton: {
    position: 'absolute',
    right: COMMON_MARGIN,
    top: COMMON_MARGIN
  },
  title: {
    paddingVertical: COMMON_PADDING,
    paddingHorizontal: COMMON_PADDING
  },
  cancelButton: {
    paddingBottom: COMMON_PADDING
  },
  description: {
    paddingHorizontal: 24,
    paddingBottom: COMMON_MARGIN,
    borderColor: darkGray
  }
});
export default styles;
