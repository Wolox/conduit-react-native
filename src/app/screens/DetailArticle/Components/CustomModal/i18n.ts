import i18next from 'i18next';

i18next.addResources('en', 'CUSTOM_COMMENT_MODAL', {
  TITLE: 'Are you sure to delete this comment?',
  CANCEL: 'Cancel',
  DELETE: 'Delete'
});
