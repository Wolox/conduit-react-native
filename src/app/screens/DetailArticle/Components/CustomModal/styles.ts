import { darkGray, grayTransparent, green, scarletRed, white } from '@constants/colors';
import { SPACING } from '@constants/dimentions';
import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
    backgroundColor: grayTransparent
  },
  modalView: {
    margin: 20,
    backgroundColor: white,
    borderRadius: 20,
    padding: 35,
    width: '93%',
    height: 180,
    alignItems: 'center',
    shadowColor: darkGray,
    shadowOffset: {
      width: SPACING.NONE,
      height: SPACING.XXXSMALL
    },
    shadowOpacity: 0.3,
    shadowRadius: SPACING.XXSMALL,
    elevation: 5
  },
  button: {
    borderRadius: 20,
    padding: 15,
    elevation: SPACING.XXXSMALL
  },
  buttonClose: {
    backgroundColor: green
  },
  buttonRemove: {
    backgroundColor: scarletRed
  },
  textStyle: {
    color: white,
    fontWeight: 'bold',
    textAlign: 'center'
  },
  modalText: {
    marginBottom: 15,
    textAlign: 'center'
  },
  containerButtons: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    width: '100%',
    marginTop: 20
  }
});
export default styles;
