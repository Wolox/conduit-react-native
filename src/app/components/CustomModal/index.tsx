import React, { ReactNode } from 'react';
import { View, TouchableWithoutFeedback, StyleProp, ViewStyle } from 'react-native';
import CustomText from '@components/CustomText';

import styles from './styles';

interface Props {
  body: ReactNode;
  title: string;
  titleComplement?: string;
  description?: string | Array<string>;
  numberLineArrayBold?: number;
  handleCloseModal?: () => void;
  style?: StyleProp<ViewStyle>;
}

const CustomModal = ({ body, title, titleComplement, description, style, numberLineArrayBold }: Props) => {
  return (
    <TouchableWithoutFeedback>
      <View style={[styles.container, style]}>
        <CustomText center style={styles.title}>
          <CustomText green bold big>
            {title}
          </CustomText>
          {titleComplement && <CustomText darkBlue>{titleComplement}</CustomText>}
        </CustomText>
        {description && !Array.isArray(description) && (
          <CustomText center small style={styles.description}>
            {description}
          </CustomText>
        )}
        {description && Array.isArray(description) && (
          <CustomText center style={styles.description}>
            {description.map((text: string, i: number) => (
              <CustomText key={i} small bold={i === numberLineArrayBold}>
                {text}
              </CustomText>
            ))}
          </CustomText>
        )}
        {body}
      </View>
    </TouchableWithoutFeedback>
  );
};

export default CustomModal;
