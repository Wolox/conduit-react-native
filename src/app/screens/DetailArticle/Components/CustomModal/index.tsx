import React from 'react';
import { View, Text, Modal, Pressable } from 'react-native';

import styles from './styles';

interface Props {
  showModal: boolean;
  setShowModal: Function;
}
export default function CustomModal({ showModal, setShowModal }: Props) {
  return (
    <Modal animationType="fade" visible={showModal} transparent={true}>
      <View style={styles.centeredView}>
        <View style={styles.modalView}>
          <Text style={styles.modalText}>Hello World!</Text>
          <View style={styles.containerButtons}>
            <Pressable style={[styles.button, styles.buttonClose]} onPress={() => setShowModal(false)}>
              <Text style={styles.textStyle}>Cancelar</Text>
            </Pressable>
            <Pressable style={[styles.button, styles.buttonRemove]} onPress={() => null}>
              <Text style={styles.textStyle}>Borrar</Text>
            </Pressable>
          </View>
        </View>
      </View>
    </Modal>
  );
}
