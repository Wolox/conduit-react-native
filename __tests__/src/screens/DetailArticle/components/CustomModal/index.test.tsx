import React from 'react';
import renderer from 'react-test-renderer';
import CustomModal from '@screens/DetailArticle/components/CustomModal';

describe('CustomModal', () => {
  test('CustomModal Snapshot', () => {
    const item = { showModal: true, setShowModal: () => null, idComment: 1, articleSlug: '' };
    const tree = renderer.create(<CustomModal {...item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
