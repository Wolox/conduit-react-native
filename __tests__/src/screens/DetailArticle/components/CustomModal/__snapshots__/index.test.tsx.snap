// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomModal CustomModal Snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000033",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 22,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 20,
          "elevation": 5,
          "height": 180,
          "margin": 20,
          "padding": 35,
          "shadowColor": "#555555",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 4,
          "width": "93%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#000",
              "fontFamily": "Nunito",
              "fontSize": 25.142857142857142,
              "fontStyle": "normal",
              "fontWeight": "400",
            },
            Array [
              Object {
                "color": "#000",
                "fontFamily": "Nunito",
                "fontSize": 25.142857142857142,
                "fontStyle": "normal",
                "fontWeight": "700",
              },
            ],
            Object {
              "marginBottom": 15,
              "textAlign": "center",
            },
          ]
        }
      >
        CUSTOM_COMMENT_MODAL:TITLE
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "marginTop": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "padding": 15,
              },
              Object {
                "backgroundColor": "#5CB85C",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            CUSTOM_COMMENT_MODAL:CANCEL
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "padding": 15,
              },
              Object {
                "backgroundColor": "#FF2400",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            CUSTOM_COMMENT_MODAL:DELETE
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
